Searc words

AleksCoder88 opened the pull request.

AleksCoder88 committed:
Создан класс для поиска слов в файле

AleksCoder88 committed:
изменения в классе поиск

AleksCoder88 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Поиск слов в файле](https://trello.com/c/xwzHan1C/3-%D0%BF%D0%BE%D0%B8%D1%81%D0%BA-%D1%81%D0%BB%D0%BE%D0%B2-%D0%B2-%D1%84%D0%B0%D0%B9%D0%BB%D0%B5)

ArhiMurder reviewed (approved):
Всё хорошо было сделано. Ошибок не замечено

anst-foto commented on ReplacWords.Lib/Search.cs:
```diff
@@ -0,0 +1,50 @@
+    class Search
+    {
+        const int size = 10;
+        string[] forbiddenWords = new string[size];
```
- Инициализация производится в конструкторе.
- Нет модификаторов доступности

anst-foto commented on ReplacWords.Lib/Search.cs:
```diff
@@ -0,0 +1,50 @@
+        const int size = 10;
+        string[] forbiddenWords = new string[size];
+        static string docsPath =
+             Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
```
Путь не определяется в классе, а передаётся через конструктор.

anst-foto commented on ReplacWords.Lib/Search.cs:
```diff
@@ -0,0 +1,50 @@
+        public void StartSearch()
+        {
+
+            var Files = Directory.EnumerateFiles(docsPath, "*.txt", SearchOption.AllDirectories);
```
Проверять можно не только файлы с расширением `*.txt`

anst-foto commented on ReplacWords.Lib/Search.cs:
```diff
@@ -0,0 +1,50 @@
+
+                string line;
+                StreamReader f = new StreamReader(currentFile);
+                while ((line = f.ReadLine()) != null)
```
Не обязательно считывать построчно, можно сразу весь файл считать, и потом искать

anst-foto reviewed (changes requested):
Учесть комментарии к тексту программы.
Постараться код поска искомых слов выделить в отдельную функцию - большая вложенность

AleksCoder88 committed:
Учтены комментарии к тексту программы.
Код поиска искомых слов выделен в отдельную функцию.

Пока осталось разобраться с тем, чтобы просматривались не только файлы ТХТ

anst-foto commented on ReplacWords.Lib/Search.cs:
```diff
@@ -0,0 +1,64 @@
+        public string[] forbiddenWords;
+        public string docsPath;
+        public string controlFolder;
+        //static string docsPath =
+        //     Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
+        //internal static string controlFolder = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments) + "/ForbiddenFiles";
```
Комментарии

anst-foto commented on ReplacWords.Lib/Search.cs:
```diff
@@ -0,0 +1,64 @@
+        }
+        public bool SearchingWordsInFile(string filePath)
+        {
+            using (FileStream fstream = File.OpenRead(filePath))
+            {
+                // выделяем массив для считывания данных из файла
+                byte[] buffer = new byte[fstream.Length];
+                // считываем данные
+                fstream.Read(buffer, 0, buffer.Length);
+                // декодируем байты в строку
+                string textFromFile = Encoding.Default.GetString(buffer);
+                //Console.WriteLine($"Текст из файла: {textFromFile}");
```
```suggestion
var file = File.ReadToEnd(); // Примерный код для сокращения
```
```

anst-foto reviewed (changes requested):
Переработать класс с учётом комментариев

anst-foto reviewed (commented):
Добавить асинхронные методы

KirillLagutin commented on ReplacWords.Lib/Search.cs:
```diff
@@ -0,0 +1,64 @@
+        public const int size = 10;
+        public string[] forbiddenWords;
+        public string docsPath;
+        public string controlFolder;
```
```suggestion
        public const int size = 10;
        public string[] forbiddenWords;
        public string docsPath;
        public string controlFolder;
        public static string[] ArrBannedWordFile;
```

KirillLagutin replied:
Просьба создать публичный массив файлов, и в него добавлять файлы, если в них замечены запрещённые слова.

KirillLagutin commented on ReplacWords.Lib/Search.cs:
```diff
@@ -0,0 +1,64 @@
+                }
+                return false;
+            }
+        }   
```
Просьба создать метод для добавления элементов в массив.
```suggestion
public static string AddElement(ref string[] array, string element)
{
  string temp[] = new array[array.Length+1];
  temp[array.Length] = element;
  for(int i=0; i<array.Length; i++)
  tamp[i] = array[i];
  array = temp;
}
        }   
```

KirillLagutin commented on ReplacWords.Lib/Search.cs:
```diff
@@ -0,0 +1,64 @@
+            {
+                if(SearchingWordsInFile(currentFile))
+                {
+                    string fileName = currentFile.Substring(docsPath.Length + 1);
+                    File.Copy(currentFile, Path.Combine(controlFolder, fileName));
```
Просьба добавлять fileName в массив ArrBannedWordFile.
```suggestion
                    string fileName = currentFile.Substring(docsPath.Length + 1);
                    File.Copy(currentFile, Path.Combine(controlFolder, fileName));
                    AddElement(ref ArrBannedWordFile, fileName);
```

AleksCoder88 committed:
- файл считывается сразу в строку при помощи метода ReadToEnd();
- для Кирилла полные имена файлов(пути) добавляются в массив ArrBannedWordFile[];
- добавлена возможность проверки файлов не только с расширением ТХТ, а и с другими расширениями;
- добавлен асинхронный метод StartSearchAsync();

anst-foto reviewed (approved):
Принимается
